scw/client_option.go

```diff
@@ -51,6 +52,42 @@ func WithHTTPClient(httpClient *http.Client) ClientOption {
 	}
 }
 
+// WithConfig client option configure a client with Scaleway configuration.
+func WithConfig(config scwconfig.Config) ClientOption {
+	return func(s *settings) {
+		accessKey, accessKeyExist := config.GetAccessKey()
+		secretKey, secretKeyExist := config.GetSecretKey()
+		if accessKeyExist && secretKeyExist {
```

Comment:
I know it changes the old behavior, but this should work from the moment the secret key exists.

Reply:
You speak about a case in which user compose with `WithAuth` (in which `accessKey` AND `sercretKey` are set) and `WithConfig` (in which `accessKey` OR `sercretKey` are set).